### ISSUE TITLE
Racecondition in `ResourceLoader::_thread_load_function(void *p_userdata)`

### ISSUE THREAD

**Issue by 0x0ACB** [godotengine/godot#95616]
### Tested versions

4.3-editor

### System information

Windows 10

### Issue description

With commit 5640e8adc9f80c237c8e373b68c3e2ced610578d the `thread_load_mutex` is unlocked after status is set to `THREAD_LOAD_LOADED`. However the code that is following that is still accessing the `resource` property. If a different thread calls `ResourceLoader::load_threaded_get` during that time `resource` will be set to `nullptr` in https://github.com/godotengine/godot/blob/ee363af0ed65f6ca9a906f6065d0bd9c19dcf9e4/core/io/resource_loader.cpp#L660

### Steps to reproduce

```
ResourceLoader::load_threaded_request(m_path, "", false);
....
const auto res = ResourceLoader::load_threaded_get_status(m_path);
if (res == ResourceLoader::THREAD_LOAD_LOADED)
{
    m_resource = ResourceLoader::load_threaded_get(m_path);
    finished();
    return true;
}
if (res != ResourceLoader::THREAD_LOAD_IN_PROGRESS)
{
    // need to remove the load task
    m_resource = ResourceLoader::load_threaded_get(m_path);
    m_error = String("Error loading resource {0}: {1}").format(varray(m_path, res));
    finished();
    return true;
}

```

### Minimal reproduction project (MRP)

-

**Comment by Chaosus** [comment]
cc @RandomShaper

**Comment by RandomShaper** [comment]
Fixed by #94169.
